temporalio/exceptions.py

```diff
@@ -0,0 +1,292 @@
+from enum import IntEnum
+from typing import Any, Iterable, Optional
+
+import temporalio.api.common.v1
+import temporalio.api.enums.v1
+import temporalio.api.failure.v1
+import temporalio.converter
+
+
+class TemporalError(Exception):
+    pass
+
+
+class FailureError(TemporalError):
```

Comment:
I'm fine with `FailureError` or `TemporalFailureError`. Prefer them over `ServiceError` because of the `ApplicationError` dissonance.

Reply:
Yeah that's fair. TemporalFailureError is a bit of a mouthful but maybe the most understandable

Reply:
It's in the `temporalio` namespace and I prefer not to stutter (people more often qualify their references in Python than Typescript). I only prefixed `Temporal` as the root error.